frontend/src/views/WheresMyGene/components/GeneSearchBar/components/SaveImage/index.tsx

```diff
@@ -89,100 +94,41 @@ export default function SaveImage({
     setIsOpen(false);
     setIsDownloading(true);
-    setEchartsRendererMode("svg");
-
-    await new Promise((resolve) => setTimeout(resolve, 5 * 1000));
-
-    try {
-      const heatmapNode = document.getElementById("view") as HTMLCanvasElement;
-      //(ashin): #3569 Get scrollTop to go back to place after downloading image
-      const heatmapContainer = document.getElementById(
-        HEATMAP_CONTAINER_ID
-      ) as HTMLCanvasElement;
-      heatmapContainerScrollTop = heatmapContainer?.scrollTop;
-
-      // Adding this class causes the y-axis scrolling to jump but is required for image download
-      heatmapNode.classList.add("CLONED");
-      const initialWidth = heatmapNode.style.width;
-      heatmapNode.style.width = `${
-        getHeatmapWidth(selectedGenes) + Y_AXIS_CHART_WIDTH_PX + 100
-      }px`;
-      const isPNG = fileType === "png";
-      const convertHTMLtoImage = isPNG ? toPng : toSvg;
-      const images = await Promise.all(
-        selectedTissues.map(async (tissueName) => {
-          const height = getHeatmapHeight(selectedCellTypes[tissueName]);
-
-          // Handles if whitespace is in the tissue name for the element ID
-          tissueName = tissueName.replace(/\s+/g, "-");
-
-          const imageURL = await convertHTMLtoImage(heatmapNode, {
-            backgroundColor: "white",
-            filter: screenshotFilter(tissueName),
-            height: height + X_AXIS_CHART_HEIGHT_PX + 120,
-            pixelRatio: 4,
-            width: heatmapNode.width,
-          });
-          // raw URI if only one tissue is selected
-
-          const input = isPNG
-            ? selectedTissues.length === 1
-              ? imageURL
-              : base64URLToArrayBuffer(imageURL)
-            : processSvg({
-                height,
-                svg: decodeURIComponent(imageURL.split(",")[1]),
-                tissueName,
-              });
 
-          return {
-            input,
-            name: `${tissueName}.${fileType}`,
-          };
-        })
-      );
-
-      //(thuang): #3569 Restore scrollTop position
-      heatmapNode.classList.remove("CLONED");
-      heatmapNode.style.width = initialWidth;
-      if (heatmapContainer) {
-        heatmapContainer.scrollTop = heatmapContainerScrollTop || 0;
-      }
+    const heatmapNode = document.getElementById("view") as HTMLDivElement;
+
+    // Options for the observer (which mutations to observe)
+    const config = { childList: true, subtree: true };
+
+    // Callback function to execute when mutations are observed
+    const callback = debounce(() => {
+      download();
+    }, MUTATION_OBSERVER_TIMEOUT);
+
+    const observer = new MutationObserver(callback);
+
+    /**
+     * (thuang): We only want to call download() once, AFTER the heatmap is fully rendered.
+     * We assume the heatmap is fully rendered after no more changes to the DOM tree
+     * after `MUTATION_OBSERVER_TIMEOUT` milliseconds
+     */
+    const download = debounce(
```

Comment:
debounce the MutationObserver callback function, so we only trigger download once after the DOM stops mutation